check-hashes.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# generates a hash for every file in $dir
+# if hash exists and is different, exit out
+
+# the intent of this script is that it's run between two corpus generations
+# the first run will generate the article-json files, then the sumfile
+# the second run will generate the article-json files again, then discover the
+# sum file exists from previous run and use that to check no differences exist
+
+set -e
+set -xv
+
+# trap ctrl-c and call ctrl_c()
+trap ctrl_c INT
+function ctrl_c() {
+    exit 1
+}
+
+sumfile="sums.md5"
+
+rm -rf article-json/ article-json-1/ article-json-2
+mkdir -p article-json
+
+python src/generate_article_json.py --num 15
```

Comment:
is this script meant to be used as a test or as an operations script?

Reply:
unless we discover some clever way of determining that article-json output should *not* have changed between revisions, then it's going to require two scrapes to complete. if time were not an issue, I'd scrape everything twice and compare checksums using something like this script, but for now this is just helping me test a subset of articles are identical when scraped twice (they're currently not).

Reply:
> unless we discover some clever way of determining that article-json output should not have changed between revisions, ...

we could say, if any of the following has changed, run the hash checker:
- requirements.txt
- app.cfg
- src/main.py
- src/conf.py
- src/iiif.py
- src/glencoe.py
- src/cache_requests.py
- src/utils.py

and possibly these as well:
- src/adaptor.py

Reply:
So it's for *testing*, and using a subset makes sense although the subset could be selected to try to provide a good coverage: https://github.com/elifesciences/bot-lax-adaptor/blob/develop/guinea-pigs.sh

Reply:
As in operations usage, I would err on the side of always running the hash checker during a backfill. 4000 articles as input, 4000 articles need to be converted to JSON and validated, but if only 100 actually change, only 100 requests to Lax being made rather than 4000 (and only 100 events are sent).

Reply:
(this particular script is for testing; other could use the same mechanism in operations)